File: 이진영/추가문제/22일차/요세푸스 문제 0.js
const main = () => {
    const input = require('fs').readFileSync('dev/stdin')
    .toString().trim();
    const [n, k] = input.split(" ").map(Number);

    let nums = [];
    for (let i = 1; i <= n; i++) {
        nums.push(i);
    }
    let answer = [];
    let count = 1;

    while (nums.length > 0) {
        while (count < k) {
            nums.push(nums.shift());
            count++;
        }
        answer.push(nums.shift());
        count = 1;
    }

    console.log("<" + answer.join(", ") + ">")
}

main();

File: 현세미/백준/요세푸스문제0.py
def solve(n, k):
    nums = list(range(1, n + 1))
    ans = []
    i = 0

    while nums:
        i = (i + k - 1) % len(nums)
        ans.append(nums.pop(i))

    print(f"<{', '.join(map(str, ans))}>")

n, k = map(int, input().split())
solve(n, k)
